STEER/STEERBase/AliOADBContainer.cxx

```diff
@@ -275,6 +276,37 @@ TObject* AliOADBContainer::GetObject(Int_t run, const char* def, TString passNam
   }
 }
 
+AliOADBCache* AliOADBContainer::GetObjectCache(Int_t run, const char* def, TString passName) const
+{
+  // Return cached object for given run or default if not found
+  TObject* obj = 0;
+  Int_t idx = GetIndexForRun(run, passName);
+  if (idx == -1) idx = GetIndexForRun(run); // try default pass for this run range
+  if (idx == -1) {
+    // no object found, try default
+    obj = fDefaultList->FindObject(def);
+  } else {
+    if (fArray!=0) {
+      obj = fArray->At(idx);
+    } else {
+      obj = GetObjectFromFile(gFile, run, def, passName);
```

Comment:
What happens if gFile points to a different file?

Reply:
To me this seemed some fallback that I do not fully understand, I left the logic as it was. Maybe one should remove it?